ui/src/components/admin/fundraise/AdminFundraiseContainer.js

```diff
@@ -0,0 +1,326 @@
+      setMilestones(selected.milestones);
+      setDeleted(selected.isDeleted);
+    }
+  }, [selected]);
+
+  const updateHandler = async () => {
+    try {
+      const res = await axios.put('/api/fundraise/update', {
+        id,
+        campaignName: campaignName,
+        categoryName: categoryName,
+        categoryKey: categoryKey,
+        amount: amount,
+        givers: givers,
+        milestones: milestones,
+        isDeleted: deleted,
+      });
+
+      if (res.status === 200) return true;
+    } catch (e) {
+      console.log(e.response);
+      toast({
+        description: e.response.data,
+        status: 'error',
+        duration: 5000,
+      });
+      return false;
+    }
+  };
+  const createHandler = async () => {
+    try {
+      const res = await axios.post('/api/fundraise/create', {
+        id,
+        campaignName: campaignName,
+        categoryName: categoryName,
+        categoryKey: categoryKey,
+        amount: amount,
+        givers: givers,
+        milestones: milestones,
+      });
+
+      if (res.status === 200) {
+        const { data } = res;
+        setId(data.id);
+        return true;
+      }
+    } catch (e) {
+      console.log(e.response);
+      toast({
+        description: e.response.data,
+        status: 'error',
+        duration: 5000,
+      });
+      return false;
+    }
+  };
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    if (!campaignName || campaignName.length === 0 || campaignNameCheck())
+      return;
+
+    setIsLoading(true);
+    let success = false;
+    if (id && id.length > 0) {
+      success = await updateHandler();
+    } else {
+      success = await createHandler();
+    }
+
+    if (success) {
+      toast({
+        description: 'Saved',
+        status: 'success',
+        duration: 5000,
+      });
+      await getData();
+    }
+    setIsLoading(false);
+  };
+  const resetHandler = () => {
+    setId('');
+    setCampaignName('');
+    setCategoryName('');
+    setCategoryKey('');
+    setAmount('');
+    setGivers('');
+    setMilestones('');
+    setSelected();
+  };
+  const campaignNameCheck = () => {
+    if (
+      fundraises &&
+      fundraises.some((i) => i.id !== id && i.campaignName === campaignName)
+    )
+      return true;
+
+    return false;
+  };
+  const handleAddMilestones = () => {
+    if (milestoneName && milestoneAmount && milestoneDeadline) {
+      const newMilestone = {
+        milestoneName,
+        milestoneAmount,
+        milestoneDeadline,
+      };
+      setMilestones([...milestones, newMilestone]);
+      setMilestoneName('');
+      setMilestoneAmount('');
+      setMilestoneDeadline('');
+    }
+  };
+
+  const handleRemoveMilestones = (index) => {
+    const updatedMilestones = [...milestones];
+    updatedMilestones.splice(index, 1);
+    setMilestones(updatedMilestones);
+  };
+
+  return (
+    <Container w="100%" maxW="100%">
+      <Heading as="h5" mb={5}>
+        Fund Raising Manager
```

Comment:
How are we going to add new fundraiser ya? Once I click into the table, the only way I can find to add new things is by refresh the page n directly input the empty form.

Reply:
yep thats how we do it for the rest of the managers: popup, testimonies
just keeping it consistent with the other workflows